addons/mail/models/res_users.py

```diff
@@ -50,6 +51,14 @@ def _compute_notification_type(self):
         # Special case: internal users with inbox notifications converted to portal must be converted to email users
         self.filtered_domain([('share', '=', True), ('notification_type', '=', 'inbox')]).notification_type = 'email'
 
+    def _compute_im_status(self):
+        presence_by_user = {
+            presence.user_id: presence.status
+            for presence in self.env["mail.presence"].search([("user_id", "in", self.ids)])
+        }
```

Comment:
Couldn't we define the inverse of presence in user to benefit from cache and avoid potential extra queries?